packages/medusa/src/services/price-list.ts

```diff
@@ -317,12 +312,13 @@ class PriceListService extends BaseService {
 
   async listProducts(
     priceListId: string,
-    selector = {},
+    selector: FilterableProductProps | Selector<Product> = {},
     config: FindConfig<Product> = {
       relations: [],
       skip: 0,
       take: 20,
-    }
+    },
+    requiredRelation = false
```

Comment:
comment: Same as other comment. I believe we should improve the readability of this prop

Reply:
Answered above